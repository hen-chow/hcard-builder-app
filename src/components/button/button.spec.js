import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './button';

describe('<Button />', () => {
  it('adds the provided CSS class', () => {
    const button = shallow(<Button className="this-is-a-test-class" />);

    const classWrapper = button.find('.this-is-a-test-class');

    expect(classWrapper.length).toEqual(1);
  });

  it('adds the provided id', () => {
    const button = shallow(<Button id="this-is-a-test-id" />);

    const idWrapper = button.find('#this-is-a-test-id');

    expect(idWrapper.length).toEqual(1);
  });

  it('renders provided children', () => {
    const button = shallow(<Button>This is a test button</Button>);

    expect(button.text()).toEqual('This is a test button');
  });

  it('invokes the provided onClick callback when user clicks on button', () => {
    const onClickCallback = jest.fn();
    const button = shallow(<Button onClick={onClickCallback} />);
    
    button.simulate('click');
    
    expect(onClickCallback.mock.calls.length).toEqual(1);
    expect(onClickCallback.mock.calls[0][0]).toEqual({
      buttonType: 'button',
      color: 'light-blue',
      onClick: onClickCallback,
    });
  });
  
  it('renders the expected HTML', () => {
    const onClickCallback = jest.fn();
    const button = renderer
      .create(<Button onClick={onClickCallback} />)
      .toJSON();

    expect(button).toMatchSnapshot();
  });
});
