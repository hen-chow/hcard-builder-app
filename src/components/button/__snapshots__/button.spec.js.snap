// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders the expected HTML 1`] = `
<button
  className="button button--light-blue"
  onClick={[Function]}
  type="button"
/>
`;
